tools/serverpod_cli/lib/src/generator/dart/library_generators/class_generators/repository_classes.dart

```diff
@@ -893,6 +928,8 @@ class BuildRepositoryClass {
       var relation = field.relation as ListRelationDefinition;
 
       methodBuilder
+        ..docs
```

Comment:
I hope I understood the attach/detach behavior here correctly.

Was not sure why that was provided as a special implementation, instead of just using `update(model.copyWith(referenceType: null))`. Would need to read up on the original ticket, in order to provide a more useful comment here.

Reply:
In essence, it is a convenience method for exactly that yes. But there is a special case where this method will be the only way to add a relation.

We have support for what we call an "implicit" one-to-many where you only define the many side (e.g. a list). This will add a hidden column on the other model that is only accessible in the database layer. To be able to still "attach" objects to the list in the database, you have to use this attach/detach method.